File: Javascript/tasks/depositProfit.js
function depositProfit(deposit, rate, threshold) {
    var fraction = [deposit, 1],
        year = 0;
    while (fraction[0] < fraction[1] * threshold) {
        fraction[0] *= 100 + rate;
        fraction[1] *= 100;
        year++;
    }
    return year;
}

File: tasks/depositProfit.cpp
int depositProfit(int deposit, int rate, int threshold) {

  std::vector<int> fraction(2);
  fraction[0] = deposit;
  fraction[1] = 1;
  int year = 0;

  while (fraction[0] < threshold) {
    fraction[0] += fraction[0] * rate / 100;
    year++;
  }

  return year;
}
